tests/codboz_frame_interpolation_test.c

```
@@ -0,0 +1,327 @@
+        (director_callback_fn)(uintptr_t)read32(base + DIRECTOR_FIXED_SLOT_OFFSET);
+    director_callback_fn interpolate =
+        (director_callback_fn)(uintptr_t)read32(base + DIRECTOR_INTERPOLATE_SLOT_OFFSET);
+    view_setter_fn view_setter = (view_setter_fn)(uintptr_t)(IMAGE_BASE + VIEW_SETTER_OFFSET);
+    camera_test_fn submit_camera =
+        (camera_test_fn)(uintptr_t)(IMAGE_BASE + CAMERA_TEST_TRAMPOLINE_OFFSET + 1u);
+
+    install_test_stubs(base);
+    uint8_t view_state[0x260];
+    memset(view_state, 0, sizeof(view_state));
+    write32(base + VIEW_STATE_GLOBAL_OFFSET, pointer32(view_state));
+    float *submitted = (float *)(view_state + VIEW_STATE_MATRIX_OFFSET);
```

Cppcheck (2.21.0)

[portability] Casting between unsigned char * and float * which have an incompatible binary data representation.

(invalidPointerCast)